x-pack/plugins/endpoint/public/applications/endpoint/store/alerts/action.ts

```diff
@@ -19,11 +19,27 @@ interface ServerReturnedAlertDetailsData {
 }
 
 interface ServerReturnedSearchBarIndexPatterns {
-  type: 'serverReturnedSearchBarIndexPatterns';
-  payload: IIndexPattern[];
+  readonly type: 'serverReturnedSearchBarIndexPatterns';
+  readonly payload: IIndexPattern[];
+}
+
+interface UserClosedAlert {
```

Comment:
It may be better to make all of these actions plural as in `UserClosedAlerts` because pretty soon we'll need to support bulk closing. Also, the payload could take a `string[]` in preparation for the bulk flow.